fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -452,21 +453,31 @@ public boolean visit(java.lang.String elementName, int elementIndex, Quantity qu
         if (!QUANTITY.equals(searchParamType)) {
             throw invalidComboException(searchParamType, quantity);
         }
-        if (quantity != null && quantity.getValue() != null && quantity.getValue().getValue() != null &&
-                (quantity.getCode() != null || quantity.getUnit() != null)) {
-            Parameter p = new Parameter();
-            p.setName(searchParamCode);
-            p.setValueNumber(quantity.getValue().getValue());
+        if (quantity.getValue() != null && quantity.getValue().hasValue()) {
```

Comment:
so, don't have to do null check for quantity any more?

Reply:
right, this makes it consistent with the other visit methods...they all assume a non-null argument and this should always be the case from FHIRPersistenceJDBCImpl due to the double-dispatch logic